assembl/views/api2/discussion.py

```diff
@@ -740,12 +745,21 @@ def get_visitors(request):
     use_first = asbool(request.GET.get("first", False))
     attribute = "first_visit" if use_first else "last_visit"
     visitors = []
+    from assembl.models import ProfileField
     for st in discussion.agent_status_in_discussion:
         if not getattr(st, attribute, None):
             continue
+        profile_fields = db.query(ProfileField).filter(ProfileField.discussion_id == discussion.id).filter(
+            ProfileField.agent_profile_id == st.agent_profile.id).all()
         data = {"time": getattr(st, attribute),
                 "name": (st.agent_profile.name or '').encode("utf-8"),
                 "email": (st.agent_profile.get_preferred_email() or '').encode("utf-8")}
+
+        for configurable_field in configurable_fields:
+            for profile_field in profile_fields:
```

Comment:
instead of using a inner for loop, you should create a dict profile_field_by_configurable_field_id
and use `profile_field_by_configurable_field_id[configurable_field.id]` here.